File: CodeWars/JavaScript/7Kyu/AlphabetWar.js
// URL: https://www.codewars.com/kata/59377c53e66267c8f6000027/train/javascript

function alphabetWar(fight){
  alphabet = {
    "left":{
      "w":4,
      "p": 3,
      "b": 2,
      "s": 1
    },
    "right":{
      "m":4,
      "q":3,
      "d":2,
      "z":1
    }
  }
  
  left = 0
  right = 0
  
  for (let i=0; i<fight.length; i++){
    if(fight[i] in alphabet.left){
      left += alphabet.left[fight[i]];
    }
    else if (fight[i] in alphabet.right){
      right += alphabet.right[fight[i]];
    }
  }
  
  if (right>left){
    return "Right side wins!";
  }
  else if(left===right){
    return "Let's fight again!";
  }
  else{
    return "Left side wins!";
  }
}

console.log(alphabetWar("tsqwpmsasm"));

File: CodeWars/Java/7Kyu/AlphabetWar.java
import java.util.*;

public class AlphabetWar {
  public static String alphabetWar(String fight) {
    HashMap<String, Integer> alphabet = new HashMap<String, Integer>();
    alphabet.put("w", -4);
    alphabet.put("p", -3);
    alphabet.put("b", -2);
    alphabet.put("s", -1);
    alphabet.put("m", 4);
    alphabet.put("q", 3);
    alphabet.put("d", 2);
    alphabet.put("z", 1);

    int score = 0;

    for (int i = 0; i < fight.length(); i++) {
      if (alphabet.containsKey(Character.toString(fight.charAt(i)))) {
        score += alphabet.get(Character.toString(fight.charAt(i)));
      }
    }

    if (score > 0) {
      return "Right side wins!";
    } else if (score == 0) {
      return "Let's fight again!";
    } else {
      return "Left side wins!";
    }
  }

  public static void main(String[] args) {
    System.out.println(alphabetWar("z"));
  }
}